tests/inference_with_NaN_simulator_test.py

```diff
@@ -224,3 +225,86 @@ def integrate_grid(distribution):
     assert torch.all(
         restricted_prior_probs[torch.logical_not(valid_thetas)] == 0.0
     ), "Rejected theta has non-zero probablity."
+
+
+@pytest.mark.parametrize(
+    "num_simulations, simulation_batch_size,\
+        num_workers, use_process_simulator, expect_error",
+    [
+        (0, None, 1, True, False),
+        (10, None, 1, True, False),
+        (100, 10, 1, True, False),
+        (100, None, 2, True, False),
+        (1000, 50, 4, True, False),
+        (100, 10, 2, False, True),
+    ],
+)
+def test_simulate_for_sbi(
+    num_simulations,
+    simulation_batch_size,
+    num_workers,
+    use_process_simulator,
+    expect_error,
+):
+    """Test the simulate_for_sbi function with various configurations."""
+    num_dim = 3
+    likelihood_shift = -1.0 * ones(num_dim)
+    likelihood_cov = 0.3 * eye(num_dim)
+    prior = BoxUniform(-2.0 * ones(num_dim), 2.0 * ones(num_dim))
+
+    def failing_simulator(theta):
+        if isinstance(theta, np.ndarray):
+            raise TypeError("This simulator does not support numpy arrays.")
+        return linear_gaussian(theta, likelihood_shift, likelihood_cov)
```

Comment:
this error should be raised automatically under the following scenario: If the function is called with `num_workers>1` then it will use `joblib` for multiprocessing and it will require that the simulator actually takes numpy arrays (not Tensor). If it was wrapped with `process_simulator` before, this will have been taken care of. If not, we will catch the following `TypeError` and raise a new one with the informative error message. 
Thus, I think you can remove the `expect_error` case in the pytest marks above and just  use 

```python
pytest.param(100, 10, 2, False, raise=TypeError, match="For multiprocessing, we switch to numpy arrays.*")
```

to cover that case.